walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/servlets/FindMeetingQuery.java

```diff
@@ -0,0 +1,151 @@
+        TimeRange backTime1 = TimeRange.fromStartEnd(whenEvent.end(),TimeRange.END_OF_DAY,true);
+        addFrontAndBackTimesToList(meetingDuration, frontTime1, backTime1, availableTime);
+    }
+    
+    /** Checks if available times overlap event times to make available times around the event */
+    private void checkOverlapsInPotentialTimesWithEventTimes(Collection<TimeRange> availableTime,TimeRange whenEvent,
+        long meetingDuration)
+    {
+        Collection<TimeRange> removeList = new ArrayList<TimeRange>();
+        Collection<TimeRange> addList = new ArrayList<TimeRange>();
+        for (TimeRange openTime : availableTime)
+        {
+            if(openTime.overlaps(whenEvent))
+            {
+                if(openTime.contains(whenEvent))
+                {
+                    removeList.add(openTime);
+                    TimeRange frontTime2 = TimeRange.fromStartEnd(openTime.start(),whenEvent.start(),false);
+                    TimeRange backTime2 = TimeRange.fromStartEnd(whenEvent.end(),TimeRange.END_OF_DAY,true);
+                    addFrontAndBackTimesToList(meetingDuration, frontTime2, backTime2,addList);
+                }
+                else if (whenEvent.start() < openTime.start())
+                {
+                    removeList.add(openTime);
+                    TimeRange overlapTime = TimeRange.fromStartEnd(whenEvent.end(),openTime.end(),false);
+                    if (overlapTime.duration() >= meetingDuration)
+                    {
+                        addList.add(overlapTime);
+                    }
+                }
+            }
+        }
+        availableTime.removeAll(removeList);
+        availableTime.addAll(addList);    
+    }
+
+    private Collection<TimeRange> checkIfOptionalAttendeesCanAttend(Collection<TimeRange> availableTime,TimeRange whenEvent)
+    {
+        Collection<TimeRange> timeWithOptional = new ArrayList<TimeRange>(); 
+        for(TimeRange currentOpenTime: availableTime)
+        {
+            if(!currentOpenTime.overlaps(whenEvent))
+            {
+                timeWithOptional.add(currentOpenTime);
+            }
+        }
+        if (timeWithOptional.isEmpty())
+        {
+            return availableTime;
+        }
+        return timeWithOptional;
+    }
+    /** Add open slot for potential meeting times before and after the event*/
+    private static void addFrontAndBackTimesToList(long meetingDuration, TimeRange frontTime,TimeRange backTime, Collection<TimeRange> timeRangeList)
+    {
+        if (frontTime.duration() >= meetingDuration)
+        {
+            timeRangeList.add(frontTime);
+        }
+        if (backTime.duration() >= meetingDuration)
+        {
+            timeRangeList.add(backTime);     
+        }
+    }
+
+    /** Check if at least one event attendee is on the meeting or optional attendees list*/
+    private boolean checkAttendees(Set<String> eventAttendees, Collection<String> checkAttendees)
```

Comment:
I also added this comment to a resolved thread, and I guess it didn't unresolve that thread (?) so I'm adding it again here.

checkAttendees() method name doesn't say much about what this method does. Also checkAttendees variable name doesn't clearly describe what the caller is supposed to provide.

What about renaming the method to anyAttendeeInvited(), does that fit what it does?

And maybe rename the parameter to invitedAttendees, does that fit what it does?

Also, you should mark all these `private` methods as `static`. In general, if you _can_ mark a method static, then you _should_ do it.